pkg/rtc/participant.go

```diff
@@ -410,6 +410,7 @@ func (p *ParticipantImpl) onPublisherAnswer(answer webrtc.SessionDescription) {
 		prometheus.ServiceOperationCounter.WithLabelValues("answer", "error", "write_message").Add(1)
 		return
 	}
+	p.TransportManager.SetPublisherLocalDescriptionSent(true)
```

Comment:
Do we need this for publisher pc? since ice restart are initiate by the client

Reply:
That is a good question. But, it is not just for ICE restart. JS SDK has this to cache candidates if it does not have remote description or if it is restarting ICE. So, to be safe, caching local candidates on server side till we know that local description was sent.
```
  async addIceCandidate(candidate: RTCIceCandidateInit): Promise<void> {                                                          
    if (this.pc.remoteDescription && !this.restartingIce) {                                                                       
    ┆ return this.pc.addIceCandidate(candidate);                                                                                  
    }                                                                                                                             
    this.pendingCandidates.push(candidate);                                                                                       
  }
```

Reply:
yeah, that's my question since js client already have the cache function, is the server cache for other clients that don't have cache?

Reply:
Sorry @cnderrauber, I misunderstood. Yes, this is for other clients (I do not know if other clients are caching or not). In a future version, JS SDK also may be able to remove it.